realm/realm-library/src/main/java/io/realm/internal/counters/ManagedRealmInteger.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal.counters;
```

Comment:
I would prefer that if someone creates another datatype, they put it into another package.  I would prefer that this package not become a catch-all for random datatypes.  Would you go for `io.realm.internal.datatypes.counters`?